### PR TITLE
Transactional outbox

### PR DESCRIPTION
**Pull request by guillermoap**

### CODE REVIEW

File: db/migrate/20220619221938_create_user_access_outbox.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class CreateUserAccessOutbox < ActiveRecord::Migration[7.0]
+  def change
+    create_table :user_access_outboxes, id: :uuid do |t|
```
Comment (sebaherrera07) [inline db/migrate/20220619221938_create_user_access_outbox.rb:3]: `id: :uuid` will make Rails not to generate another `id` column?

File: db/migrate/20220705174515_create_user_access_consumed_message.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateUserAccessConsumedMessage < ActiveRecord::Migration[7.0]
+  def change
+    create_table :user_access_consumed_messages do |t|
```
Comment (sebaherrera07) [inline db/migrate/20220705174515_create_user_access_consumed_message.rb:3]: Should these ones need a `id: :uuid` too?

File: register-postgres.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  "name": "user-access-outbox-connector",
+  "config": {
+    "connector.class": "io.debezium.connector.postgresql.PostgresConnector",
+    "tasks.max": "1",
+    "plugin.name": "pgoutput",
+    "database.hostname": "db",
+    "database.port": "5432",
+    "database.user": "postgres",
+    "database.password": "postgres",
+    "database.dbname" : "rails_modular_monolith_with_ddd_development",
```
Comment (sebaherrera07) [inline register-postgres.json:11]: So if this one is used for development, where is this set for production?

And for testing, we won't use any connector, we will just mock things?

File: db/migrate/20220705174515_create_user_access_consumed_message.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+      t.uuid :event_id
+      t.string :aggregate
```
Comment (sebaherrera07) [inline db/migrate/20220705174515_create_user_access_consumed_message.rb:5]: I believe we should add a composed index by these 2 columns.

File: components/user_access/spec/factories/user_access/outboxes.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+          lsn: 23382416,
+          xmin: nil
+        },
+        op: 'c',
+        ts_ms: 1656622474973,
+        transaction: nil
+      }
+    end
+    aggregate { UserAccess::UserRegistration.name }
+    identifier { SecureRandom.uuid }
```
Comment (sebaherrera07) [inline components/user_access/spec/factories/user_access/outboxes.rb:47]: I believe this column is not called `identifier` ?

File: README.md [hunk 6]
```diff
@@ -400,6 +400,26 @@ List of technologies, frameworks and libraries used for implementation:
...
...ype":"string","optional":false,"field":"id"},{"type":"int64","optional":false,"field":"total_order"},{"type":"int64","optional":false,"field":"data_collection_order"}],"optional":true,"field":"transaction"}],"optional":false,"name":"dbserver1.public.user_access_user_registrations.Envelope"},"payload":{"before":null,"after":{"id":1,"login":"angle_parisian","email":"irwin@blanda-thiel.biz","encrypted_password":"$2a$12$FUctOJz69VZsYwD1xYwHBurKvqb4CC8xVDRuxt7T6GqFAlyqgysHK","first_name":"Moises","last_name":"Kemmer","name":"Moises Kemmer","status_code":1,"registered_at":1656104074437113,"confirmed_at":null,"created_at":1656622474653699,"updated_at":1656622474653699,"confirmation_token":"a423a2da-54c1-4a4e-ab8b-f4588874be9f","confirmation_sent_at":1656536069675604,"unconfirmed_email":null,"identifier":"7d8f60e3-5e7f-4c11-b18b-5cc01ceea3da"},"source":{"version":"1.9.4.Final","connector":"postgresql","name":"dbserver1","ts_ms":1656622474660,"snapshot":"false","db":"rails_modular_monolith_with_ddd_development","sequence":"[null,\"23382416\"]","schema":"public","table":"user_access_user_registrations","txId":509,"lsn":23382416,"xmin":null},"op":"c","ts_ms":1656622474973,"transaction":null}}
```
Comment (santib) [inline README.md:421]: do we want to keep this?

**Review by santib — APPROVED** [review]

**Pull request merged**